runtime/codert_vm/decomp.cpp

```diff
@@ -1200,12 +1200,45 @@ c_jitDecompileAtExceptionCatch(J9VMThread * currentThread)
 	U_8 *jitPC = decompRecord->pc;
 	/* Simulate a call to a resolve helper to make the stack walkable */
 	buildBranchJITResolveFrame(currentThread, jitPC, J9_STACK_FLAGS_JIT_EXCEPTION_CATCH_RESOLVE);
-	/* Discard the existing decompilation in favour of a new one at the exception catch point */
+
+	J9JavaVM *vm = currentThread->javaVM;
+	J9JITDecompileState decompileState;
+	J9StackWalkState walkState;
 	J9OSRBuffer *osrBuffer = &decompRecord->osrBuffer;
 	UDATA numberOfFrames = osrBuffer->numberOfFrames;
 	J9OSRFrame *osrFrame = (J9OSRFrame*)(osrBuffer + 1);
+	J9JITExceptionTable *metaData = NULL;
+
+	/* Collect the required information from the stack - top visible frame is the decompile frame */
+	walkState.flags = J9_STACKWALK_ITERATE_FRAMES | J9_STACKWALK_SKIP_INLINES | J9_STACKWALK_VISIBLE_ONLY | J9_STACKWALK_MAINTAIN_REGISTER_MAP | J9_STACKWALK_ITERATE_HIDDEN_JIT_FRAMES | J9_STACKWALK_SAVE_STACKED_REGISTERS;
+	walkState.skipCount = 0;
+	walkState.frameWalkFunction = decompileMethodFrameIterator;
+	walkState.walkThread = currentThread;
+	walkState.userData1 = &decompileState;
+	walkState.userData2 = NULL;
+	vm->walkStackFrames(currentThread, &walkState);
+	metaData = decompileState.metaData;
+
+	/* Determine in which inlined frame the exception is being caught */
+	UDATA newNumberOfFrames = 1;
+	void *stackMap = NULL;
+	void *inlineMap = NULL;
+	void *inlinedCallSite = NULL;
+	/* Note we need to add 1 to the JIT PC here in order to get the correct map at the exception handler
+	 * because jitGetMapsFromPC is expecting a return address, so it subtracts 1.  The value stored in the
+	 * decomp record is the start address of the compiled exception handler.
+	 */
+	jitGetMapsFromPC(vm, metaData, (UDATA)jitPC + 1, &stackMap, &inlineMap);
+	Assert_CodertVM_false(NULL == inlineMap);
```

Comment:
Are we guaranteed to have an inlineMap even if nothing was inlined here?  Should this assert be moved inside the if?

Reply:
This assert has never fired, in testing or production, so I'll say it's correct.

Reply:
And GAC pointed out above (and I missed):
> The changes in c_jitDecompileAtExceptionCatch are entirely the revert of the previous attempt.